### PR TITLE
New chat service

### PR DESCRIPTION
**Pull request by gparant**

### CODE REVIEW

File: front/chat/src/Xmpp/MucRoom.ts [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+                return list;
+            });
+        }
+    }
+
+    public getPresenceStore(): UsersStore {
+        return {
+            subscribe: this.presenceStore.subscribe,
+        };
+    }
```
Comment (moufmouf) [inline front/chat/src/Xmpp/MucRoom.ts:292]: Simply casting it as a UsersStore (which is a Readable) is enough.

```suggestion
    public getPresenceStore(): UsersStore {
        return this.presenceStore;
    }
```

Typescript will do the rest :D

File: front/chat/tailwind.config.cjs [hunk 2]
```diff
@@ -0,0 +1,48 @@
+const defaultTheme = require("tailwindcss/defaultTheme");
```
Comment (moufmouf) [inline front/chat/tailwind.config.cjs:1]: Same thing, this file is probably in common with the base file

**Review by moufmouf — COMMENTED** [review]
I haven't had time to finish the review yet, but here are my first items.

File: front/chat/src/Messages/ts-proto-generated/google/protobuf/wrappers.ts [hunk 3]
```diff
@@ -0,0 +1,584 @@
+/* eslint-disable */
```
[the remark this answers is not in the document]
Comment (cesar-cardinale) [reply front/chat/src/Messages/ts-proto-generated/google/protobuf/wrappers.ts:1]: Okay it seems to be worth, but I need to gitignore /Messages/* or just /ts-proto-generated/* ?

File: front/src/Api/Events/JoinMucEvent.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import { z } from "zod";
+
+export const isJoinMucEvent = z.object({
```
Comment (gparant) [inline front/src/Api/Events/JoinMucEvent.ts:3]: @cesar-cardinale this file is used in `front/src` or already moved in `front/chat/src`?

File: front/src/Api/Events/AskPositionEvent.ts [hunk 5]
```diff
@@ -0,0 +1,11 @@
+import { z } from "zod";
+
+export const isAskPositionEvent = z.object({
```
Comment (gparant) [inline front/src/Api/Events/AskPositionEvent.ts:3]: @cesar-cardinale this file is used in `front/src` or already moved in `front/chat/src`?

File: front/src/Api/Events/LeaveMucEvent.ts [hunk 6]
```diff
@@ -0,0 +1,10 @@
+import { z } from "zod";
+
+export const isLeaveMucEvent = z.object({
```
Comment (gparant) [inline front/src/Api/Events/LeaveMucEvent.ts:3]: @cesar-cardinale this file is used in `front/src` or already moved in `front/chat/src`?

File: front/src/Components/Chat/ChatCircleDiscussion.svelte [hunk 7]
```diff
@@ -0,0 +1,183 @@
+<script lang="ts">
```
Comment (gparant) [inline front/src/Components/Chat/ChatCircleDiscussion.svelte:1]: @cesar-cardinale this file is used in `front/src` or already moved in `front/chat/src`?

File: front/src/Components/Chat/ChatForum.svelte [hunk 8]
```diff
@@ -0,0 +1,33 @@
+<script>
```
Comment (gparant) [inline front/src/Components/Chat/ChatForum.svelte:1]: @cesar-cardinale this file is used in `front/src` or already moved in `front/chat/src`?

File: front/src/Components/Chat/ChatUser.svelte [hunk 9]
```diff
@@ -0,0 +1,36 @@
+<script>
```
Comment (gparant) [inline front/src/Components/Chat/ChatUser.svelte:1]: @cesar-cardinale this file is used in `front/src` or already moved in `front/chat/src`?

File: front/src/Xmpp/XmppClient.ts [hunk 10]
```diff
@@ -5,75 +5,20 @@ import type { RoomConnection } from "../Connexion/RoomConnection";
 import { mucRoomsStore, xmppServerConnectionStatusStore } from "../Stores/MucRoomsStore";
 import type { MucRoomDefinitionInterface } from "../Network/ProtobufClientUtils";
 import ElementExt from "./Lib/ElementExt";
-import { XmppConnectionStatusChangeMessage_Status as Status } from "../Messages/ts-proto-generated/protos/messages";
 
 export class XmppClient {
```
Comment (gparant) [inline front/src/Xmpp/XmppClient.ts:9]: @cesar-cardinale if useless, delete this?

File: front/src/Xmpp/MucRoom.ts [hunk 11]
```diff
@@ -7,7 +7,7 @@ import { writable } from "svelte/store";
 import ElementExt from "./Lib/ElementExt";
```
Comment (gparant) [inline front/src/Xmpp/MucRoom.ts:7]: @cesar-cardinale if useless, delete this?

**Pull request merged**